api-server-nest/src/announcements/dto/create-or-update-announcement-request.ts

```diff
@@ -1,16 +1,16 @@
 import isNotNullOrUndefined from "../../utils/is-not-null-or-undefined";
 import cloneAttributes from "../../utils/clone-attributes";
 import {ApiProperty} from "@nestjs/swagger";
-import {IsNotEmpty} from "class-validator";
+import {IsOptional} from "class-validator";
 
 export class CreateOrUpdateAnnouncementRequest {
-    constructor(copy?: CreateOrUpdateAnnouncementRequest) {
+    constructor(copy?: Partial<CreateOrUpdateAnnouncementRequest>) {
         if (isNotNullOrUndefined(copy)) {
             cloneAttributes<CreateOrUpdateAnnouncementRequest>(copy, this)
         }
     }
 
     @ApiProperty({example: 'Life is not a dream. Beware. And beware. And beware.'})
-    @IsNotEmpty()
+    @IsOptional()
     message: string
 }
```

Comment:
We want to allow this to be empty for the initial case where we create a blank announcement.

FYI, I hit a strange bug where once I removed IsNotEmpty but before I added IsOptional the message was not coming through to to my controller method. I'm not sure if that's a bug in the framework or if some type of decorator is necessary for the validatedTransformPipe we are using, something to watch out for.